### ISSUE TITLE
Make input and output generic/minimize code duplication

### ISSUE THREAD

**Issue by BlackPhlox** [BlackPhlox/bevy_midi#15]
Something like this can also be implemented for input, maybe we can do something more generic for both of them?

_Originally posted by @BlackPhlox in https://github.com/BlackPhlox/bevy_midi/pull/12#discussion_r939655190_

**Comment by BlackPhlox** [comment]
Yeah, generic was the wrong word here. End goal is to minimize code duplication, not made generic.

**Comment by grace125** [comment]
Ah okay good, glad we're on the same page :)
